### ISSUE TITLE
Code 3 Invalid Argument_Invalid user

### ISSUE THREAD

**Issue by martinw2024** [jifeline-networks/openOBD#6]
Hello Frank,

since today i got the GRPC code 3, even with a blank OBD template on replit. My Credentials didnt changed.

any idea?

BR

Martin

**Comment by Frank-openOBD** [comment]
We try to reproduce this error, but we could not. Can you give us a little bit more details of what you are trying to do?

Code3 = 
The client specified an invalid argument. Note that this differs from FAILED_PRECONDITION. INVALID_ARGUMENT indicates arguments that are problematic regardless of the state of the system (e.g., a malformed file name).

**Comment by martinw2024** [comment]
SOLVED:

with pip install -> openobd 1.15.2 installs openobd-protocol 1.15.2 and not 1.16.0. 
I upgraded by myself to 1.16.0, now its works again.

chardet           5.2.0
et-xmlfile       1.1.0
grpcio           1.60.1
grpcio-tools     1.60.1
numpy            1.26.2
openobd          1.15.2
openobd-protocol 1.16.0
openobd-utils    1.15.2
openpyxl         3.1.2
pandas           2.1.4
Pillow           10.1.0
pip              23.2.1
protobuf         4.25.5
python-dateutil  2.8.2
python-dotenv    1.0.1
pytz             2023.3.post1
reportlab        4.2.2
setuptools       75.1.0
six              1.16.0
tzdata           2023.

___________________________________________


yes, basically I working in VSC. I put in the credentials within .env file like mentioned on your website. I always got code 3 since then.

To recheck that my credentials are still working i switched back to replit and put my secrets under tools like usual. But when i let the demo run i get now: 

[i] Starting openOBD session
 [!] Error: Something unexpected occurred [OpenOBDException: invalid_client (gRPC status: 3, invalid argument)]
Traceback (most recent call last):
  File "/home/runner/openOBD-Demo/.pythonlibs/lib/python3.11/site-packages/openobd/openobd.py", line 42, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/openOBD-Demo/.pythonlibs/lib/python3.11/site-packages/openobd/openobd.py", line 301, in _get_session_controller_token
    response = self.session_controller.getSessionControllerToken(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/openOBD-Demo/.pythonlibs/lib/python3.11/site-packages/grpc/_channel.py", line 1160, in __call__
    return _end_unary_response_blocking(state, call, False, None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/openOBD-Demo/.pythonlibs/lib/python3.11/site-packages/grpc/_channel.py", line 1003, in _end_unary_response_blocking
    raise _InactiveRpcError(state)  # pytype: disable=not-instantiable
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
    status = StatusCode.INVALID_ARGUMENT
    details = "invalid_client"
    debug_error_string = "UNKNOWN:Error received from peer  {created_time:"2024-09-23T09:15:39.397065003+00:00", grpc_status:3, grpc_message:"invalid_client"}"
>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/openOBD-Demo/client.py", line 19, in <module>
    openobd = OpenOBD()
              ^^^^^^^^^
  File "/home/runner/openOBD-Demo/.pythonlibs/lib/python3.11/site-packages/openobd/openobd.py", line 222, in __init__
    self.session_controller = OpenOBDSessionController(**kwargs)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/openOBD-Demo/.pythonlibs/lib/python3.11/site-packages/openobd/openobd.py", line 280, in __init__
    self._get_session_controller_token()
  File "/home/runner/openOBD-Demo/.pythonlibs/lib/python3.11/site-packages/openobd/openobd.py", line 46, in wrapper
    raise OpenOBDException(details=e.details(), status=e.code().value[0], status_description=e.code().value[1])
openobd.openobd_exceptions.OpenOBDException: OpenOBDException: invalid_client (gRPC status: 3, invalid argument)


Normally I got something like "ticket id not there" (because i dont run a ticket everytime to connect to beacuse im lazy) but now its code 3. Its since the update to 1.15.2

FYI im using the V2 API to call ticket data like usual and it works fine.

**Comment by Frank-openOBD** [comment]
We have strong indication that there is a mismatch between client API key, client ID and client secret. Please make sure you are using the correct and matching ones.

**Comment by martinw2024** [comment]
see my EDIT, I solved the issue.

SOLVED:

with pip install -> openobd 1.15.2 installs openobd-protocol 1.15.2 and not 1.16.0.
I upgraded by myself to 1.16.0, now its works again.

chardet 5.2.0
et-xmlfile 1.1.0
grpcio 1.60.1
grpcio-tools 1.60.1
numpy 1.26.2
openobd 1.15.2
openobd-protocol 1.16.0
openobd-utils 1.15.2
openpyxl 3.1.2
pandas 2.1.4
Pillow 10.1.0
pip 23.2.1
protobuf 4.25.5
python-dateutil 2.8.2
python-dotenv 1.0.1
pytz 2023.3.post1
reportlab 4.2.2
setuptools 75.1.0
six 1.16.0
tzdata 2023.
